src/util/isNil.ts

```diff
@@ -0,0 +1,3 @@
+export function isNil(value: unknown) {
+  return value == null;
+}
```

Comment:
so TypeScript can be weird with this. Does this know that `== null` can be undefined or null? Maybe a better way to check this would be to see if it type narrows for something that can be both undefined and null

Reply:
We could just make it a type guard as well.

Reply:
Cool, that works for me

Reply:
If it was me I would add unit and type tests too - first to show that indeed yes yes double equal does check for undefined. There will always be some people who are not sure about that bit. And second test to show that this function allows use inside an `Array.prototype.filter` call; it will need the typeguard to work that way.

Reply:
Thank you Pavel, I will do that. That is exactly why I asked you for PR review as well. ❤️ Appreciate it.

Reply:
Sometimes when a function that validates a type (this helper) is within the scope of another function that does something (filter) TypeScript isn't able type narrow correctly.

Here is this function as an example https://www.typescriptlang.org/play?ts=4.7.4#code/FAjGHsDsGcBcAICW0ByiA28C88AUA3AQ3QFcBTALnhMgGtJwB3SASmwD54BvYeeAJzKwS-SPCKky2HJBLp0AbmABfMFDjxC-flVxx+iSAHN4AH2qQAJmQBmhspbPxZ8lgG0AutnhuA5L4AaZzl0DyVgAHoI+Dt0WDJBR1BCMTgMTAAjKRdMcH4LaztIB3gyfDIxWAALcBIjKvhGKVAqslBaGLykWGAIGARY+MTvLX4AOkGEgmIObl4BIRExAEJkNHRp9BYlZRYgA

Reply:
So from a helper like this I have two expectations:

1. the runtime behaviour - you have that covered well in your tests @nikolasrieble 
2. and the compile time behaviour - that's missing from tests. @ckifer 's example is one of them, I would add another:

https://www.typescriptlang.org/play?ts=5.2.2#code/MYewdgzgLgBAlhAcnANjAvDAFANwIYoCuApgFwyFgDWYIA7mAJTn5HHwQxiEpoA+FMABNiAMzhhiQjAD4YAbwBQMGACdiUQqrAxWJDJm68A3IoC+ixaMrAoccDFFgszGNFUSA5jAFG08tQ0tHQByEJgLRVBIWFZydy8fLh40TCcXSwB6TJgoAAsENzyQHmkAI3Z81XpEvB1iVWrVKPBoeDBWOCF4qA8wb0xWU0U4UWwAQgRkFFwCRkYFZRhs3ILOCGLSrhBYKvol6Lb1cUlut17EwYJTCyA

Reply:
To fix the array filter example we would need another helper: https://www.typescriptlang.org/play?ts=5.2.2#code/GYVwdgxgLglg9mABDAzgEQKbBmDATRAHgBUA+ACgDcBDAGxAwC5FjEAfRcPLHfdxMCFq0AlMxr0MyFC0QBvAFCJEAJwxQQKpBIaIAhAF4BQ2gG4FAXwUKICFFETUVK5uXsqcAc35ceuAhyCwiIA2gC6iEYhAOTRADTGwmHmCgD0qYgA7lIQ1Ehq2LiOSE4q1ACeAnCZNnYO2LRQGGp4zO5e4ZGOzgB0DU0q5KiYhfgiCkA